From d624ceb7f2e6e286cf876f2ff9bf90532166c7f9 Mon Sep 17 00:00:00 2001
From: Franck Bui <fbui@suse.com>
Date: Fri, 12 Jan 2024 10:54:57 +0100
Subject: [PATCH 5001/5002] Revert "udev: update devlink with the newer device
 node even when priority is equivalent"

This reverts commit 277cb24ffb7a520f640eb36d11340f11bf0c7c4f.
---
 src/udev/udev-node.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/udev/udev-node.c b/src/udev/udev-node.c
index e12c26ce5a..6ba38087dc 100644
--- a/src/udev/udev-node.c
+++ b/src/udev/udev-node.c
@@ -467,13 +467,13 @@ static int link_update(sd_device *dev, const char *slink, bool add) {
                                 /* The devlink priority is downgraded. Another device may have a higher
                                  * priority now. Let's find the device node with the highest priority. */
                         } else {
-                                if (current_prio > prio)
-                                        /* The devlink with a higher priority already exists and is owned by
-                                         * another device. Hence, it is not necessary to recreate it. */
+                                if (current_prio >= prio)
+                                        /* The devlink with equal or higher priority already exists and is
+                                         * owned by another device. Hence, it is not necessary to recreate it. */
                                         return 0;
 
-                                /* This device has the equal or a higher priority than the current. Let's
-                                 * create the devlink to our device node. */
+                                /* This device has a higher priority than the current. Let's create the
+                                 * devlink to our device node. */
                                 return node_symlink(dev, NULL, slink);
                         }
 
-- 
2.35.3

